sdcm/tester.py

```diff
@@ -787,19 +788,30 @@ def init_resources(self, loader_info=None, db_info=None,
         elif cluster_backend == 'baremetal':
             self.get_cluster_baremetal()
 
-        seeds_num = self.params.get('seeds_num', default=1)
-        if self.params.get('instance_provision') == 'mixed':
-            for node in self.db_cluster.nodes:
-                if seeds_num == 0:
-                    break
-                if node.is_spot:
-                    node.is_seed = True
-                    seeds_num -= 1
+    def set_seeds(self):
+        seeds_selector = self.params.get('seeds_selector')
+        seeds_num = self.params.get('seeds_num')
+
+        if seeds_selector == 'reflector':
+            node = self.db_cluster.nodes[0]
+            node.wait_ssh_up()
+            # When cluster just started, seed IP in the scylla.yaml may be '127.0.0.1'
+            # In this case we want ignore it and wait, when reflector will select real node and update scylla.yaml
+            self.db_cluster.seed_nodes_ips = wait.wait_for(self.db_cluster.get_seed_selected_by_reflector,
+                                                           step=10, text='Waiting for seed is selected by reflector',
+                                                           timeout=300, throw_exc=True)
         else:
-            for i in range(seeds_num):
-                self.db_cluster.nodes[i].is_seed = True
-                if self.cs_db_cluster:
-                    self.cs_db_cluster.nodes[i].is_seed = True
+            if seeds_selector == 'random':
+                selected_nodes = random.sample(self.db_cluster.nodes, seeds_num)
+            else:
```

Comment:
add a comment in the code that it is seed "first" strategy or make it explicitly with `elif seeds_selector=='first: .... else raise Exception("unexpected seed selector %s" % seeds_selector)'`